### PR TITLE
Winsock updates, iocp calls

### PR DESCRIPTION
**Pull request by jon-lipstate**
Updates i think cover IOCP and several additional winsock2 use cases.

I was not certain where the structs go, let me know if i need to move them.

### CODE REVIEW

**Comment by jon-lipstate** [comment]
@Kelimion Ready for review

File: core/sys/windows/kernel32.odin [hunk 1]
```diff
@@ -382,356 +191,195 @@ foreign kernel32 {
...
-}
-
 
-SECTION_QUERY                :: DWORD(0x0001)
-SECTION_MAP_WRITE            :: DWORD(0x0002)
-SECTION_MAP_READ             :: DWORD(0x0004)
-SECTION_MAP_EXECUTE          :: DWORD(0x0008)
-SECTION_EXTEND_SIZE          :: DWORD(0x0010)
-SECTION_ALL_ACCESS           :: STANDARD_RIGHTS_REQUIRED | SECTION_QUERY | SECTION_MAP_WRITE | SECTION_MAP_READ | SECTION_MAP_EXECUTE | SECTION_EXTEND_SIZE
+	// [MS-Docs](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setfilecompletionnotificationmodes)
```
Comment (Kelimion) [inline core/sys/windows/kernel32.odin:212]: Maybe you could just add these entries and leave the rest of `kernel32.odin` as-is.

File: core/sys/windows/kernel32.odin [hunk 2]
```diff
@@ -3,75 +3,53 @@ package sys_windows
...
-FOREGROUND_GREEN           :: WORD(0x0002)
-FOREGROUND_RED             :: WORD(0x0004)
-FOREGROUND_INTENSITY       :: WORD(0x0008)
-BACKGROUND_BLUE            :: WORD(0x0010)
-BACKGROUND_GREEN           :: WORD(0x0020)
-BACKGROUND_RED             :: WORD(0x0040)
-BACKGROUND_INTENSITY       :: WORD(0x0080)
-COMMON_LVB_LEADING_BYTE    :: WORD(0x0100)
-COMMON_LVB_TRAILING_BYTE   :: WORD(0x0200)
+FOREGROUND_BLUE :: WORD(0x0001)
```
Comment (Kelimion) [inline core/sys/windows/kernel32.odin:6]: What happened to `kernel32.odin`? It's had its formatting screwed up on lines entirely unrelated to this PR.

**Review by Kelimion — CHANGES_REQUESTED** [review]

**Comment by Kelimion** [comment]
You seem to have changed `kernel32.odin` a bit too much with reformatted lines that have nothing to do with IOCP. Git can't even merge the patch as a result of it.

File: core/sys/windows/kernel32.odin (hunk 1 above)
Comment (jon-lipstate) [reply core/sys/windows/kernel32.odin:408]: apologies, i think i opened it from an editor that had ols running, and it autoformatted when i made an edit. i believe the edits are cleaned up.

**Comment by Kelimion** [comment]
Looks good to me.

**Pull request merged**